src/network/networkd-link.c

```diff
@@ -1712,6 +1746,14 @@ void link_drop(Link *link) {
 
         log_link_debug(link, "Link removed");
 
+        if (link->alternative_names) {
+                char **n;
+
+                STRV_FOREACH(n, link->alternative_names)
+                        hashmap_remove_value(link->manager->links_by_name, *n, link);
```

Comment:
Why do you use `hashmap_remove_value()`? It seems `hashmap_remove()` is sufficient.

Reply:
See https://github.com/systemd/systemd/pull/17248#discussion_r507828602

Reply:
Thank you for the reference. OK, let's keep this as is.